pkg/controller/core/queuedworkload_controller.go

```diff
@@ -142,27 +152,39 @@ func (r *QueuedWorkloadReconciler) Update(e event.UpdateEvent) bool {
 	case prevStatus == pending && status == pending:
 		if !r.queues.UpdateWorkload(oldWl, wl.DeepCopy()) {
 			log.V(2).Info("Queue for updated workload didn't exist; ignoring for now")
+			if err := r.UpdateWorkloadStatus(context.Background(), wl, kueue.QueuedWorkloadAdmitted, pending,
+				fmt.Sprintf("Queue: %s for workload didn't exist", oldWl.Spec.QueueName)); err != nil {
+				log.V(2).Error(err, "Updating QueuedWorkload Status")
+			}
 		}
 
 	case prevStatus == pending && status == admitted:
 		r.queues.DeleteWorkload(oldWl)
 		if !r.cache.AddOrUpdateWorkload(wl.DeepCopy()) {
 			log.V(2).Info("ClusterQueue for workload didn't exist; ignored for now")
 		}
+		if err := r.UpdateWorkloadStatus(context.Background(), wl, kueue.QueuedWorkloadAdmitted, admitted,
```

Comment:
as discussed below, I think we shouldn't set this (if we wanted to set it, it would actually be in the scheduler not here).